lib/elixir/src/elixir_parser.yrl

```diff
@@ -778,21 +778,42 @@ build_map_update(Left, {Pipe, Struct, Map}, Right, Extra) ->
 
 %% Blocks
 
-build_block(Exprs) -> build_block(Exprs, []).
+build_block(Exprs) -> build_block(Exprs, {}).
 
-build_block([{unquote_splicing, _, [_]}]=Exprs, Meta) ->
-  {'__block__', Meta, Exprs};
-build_block([Expr], _Meta) ->
+build_block([{unquote_splicing, _, [_]}]=Exprs, BeforeAfter) ->
+  {'__block__', block_meta(BeforeAfter), Exprs};
+build_block([{Op, ExprMeta, Args}], {Before, After}) ->
+  ExprMetaWithExtra =
+    case ?token_metadata() of
+      true ->
+        ParensEntry = meta_from_token(Before) ++ [{closing, meta_from_token(After)}],
+        case ExprMeta of
+          % If there are multiple parens, those will result in subsequent
+          % build_block/2 calls, so we can assume parens entry is first
+          [{parens, Parens} | Meta] ->
+            [{parens, [ParensEntry | Parens]} | Meta];
```

Comment:
@josevalim with this approach the entry will be first, as in `[parens: ..., line: ..., column: ...]`. If we want to append it, then we will need to iterate the entries (in case there's a `parens` already). Let me know which one you prefer :)